### ISSUE TITLE
Modernize the loading indicators for enterprise accounts

### ISSUE THREAD

**Issue by pixdemo** [KeypDemoCorp/awesome-ui#354]
We should add logging so we can debug this more easily in the future.

The retry logic needs to be more resilient to transient failures.

Accessibility should be considered when implementing this.

**Comment by pixdemo** [comment]
Let's sync on this in the next standup to align on the approach.
